File: classifier/data/raw/ai/samples/project_euler_zig/problem_003.zig
const std = @import("std");

// Project Euler Problem 3: find the largest prime factor of 600851475143.
pub fn main() !void {
    var number: u64 = 600_851_475_143;
    var factor: u64 = 2;
    var largest: u64 = 1;

    while (factor <= number / factor) {
        while (number % factor == 0) {
            largest = factor;
            number /= factor;
        }
        factor += if (factor == 2) 1 else 2;
    }
    if (number > 1) largest = number;
    try printNumber(largest);
}

fn printNumber(number: anytype) !void {
    var buffer: [32]u8 = undefined;
    const output = try std.fmt.bufPrint(&buffer, "{d}\n", .{number});
    try std.Io.File.writeStreamingAll(.stdout(), std.Io.Threaded.global_single_threaded.io(), output);
}

File: classifier/data/raw/ai/samples/project_euler_java/Problem003.java
/** Project Euler Problem 3: largest prime factor of 600851475143. */
public class Problem003 {
    public static void main(String[] args) {
        long value = 600_851_475_143L;
        long largest = 1;
        for (long factor = 2; factor * factor <= value; factor += factor == 2 ? 1 : 2) {
            while (value % factor == 0) {
                largest = factor;
                value /= factor;
            }
        }
        if (value > 1) largest = value;
        System.out.println(largest);
    }
}
